### ISSUE TITLE
Machinemasterkeys fails on some Recovery files

### ISSUE THREAD

**Issue by leechristensen** [GhostPack/SharpDPAPI#6]
```
C:\>C:\tools\SharpDPAPI\SharpDPAPI\bin\Debug\SharpDPAPI.exe machinemasterkeys
  __                 _   _       _ ___
 (_  |_   _. ._ ._  | \ |_) /\  |_) |
 __) | | (_| |  |_) |_/ |  /--\ |  _|_
                |
  v1.6.1


[*] Action: Machine DPAPI Masterkey File Triage

[*] Elevating to SYSTEM via token duplication for LSA secret retrieval
[*] RevertToSelf()

[*] Secret  : DPAPI_SYSTEM
[*]    full: <snip>
[*]    m/u : <snip>


[!] Unhandled SharpDPAPI exception:

System.UnauthorizedAccessException: Access to the path 'C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat{86283df8-8d88-4a51-957d-449ff1919746}.TM.blf' is denied.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.File.InternalReadAllBytes(String path, Boolean checkHost)
   at SharpDPAPI.Triage.TriageSystemMasterKeys(Boolean show) in C:\tools\SharpDPAPI\SharpDPAPI\lib\Triage.cs:line 161
   at SharpDPAPI.Commands.Machinemasterkeys.Execute(Dictionary`2 arguments) in C:\tools\SharpDPAPI\SharpDPAPI\Commands\Machinemasterkeys.cs:line 15
   at SharpDPAPI.Domain.CommandCollection.ExecuteCommand(String commandName, Dictionary`2 arguments) in C:\tools\SharpDPAPI\SharpDPAPI\Domain\CommandCollection.cs:line 46
   at SharpDPAPI.Program.Main(String[] args) in C:\tools\SharpDPAPI\SharpDPAPI\Program.cs:line 24
```

```
FullName                                                                                                                                     
--------                                                                                                                                     
C:\Windows\System32\Microsoft\Protect\Recovery                                                                                               
C:\Windows\System32\Microsoft\Protect\S-1-5-18                                                                                               
C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat                                                                                  
C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat.LOG1                                                                             
C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat.LOG2                                                                             
C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat{440de6b5-c574-4468-b756-a0b8ea25fdc0}.TM.blf                                     
C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat{ea68be95-bae7-405a-8b69-51103f0bfa0d}.TMContainer00000000000000000001.regtrans-ms
C:\Windows\System32\Microsoft\Protect\Recovery\Recovery.dat{7c30ad6c-e65d-42c6-b7a6-04acffb7cb1f}.TMContainer00000000000000000002.regtrans-ms
C:\Windows\System32\Microsoft\Protect\S-1-5-18\User                                                                                          
C:\Windows\System32\Microsoft\Protect\S-1-5-18\d60dafcc-6ea7-4d5c-a955-c2ab0e6017b6  
C:\Windows\System32\Microsoft\Protect\S-1-5-18\b32c164f-b24a-414b-98b5-0a6640eab86c                                                          
C:\Windows\System32\Microsoft\Protect\S-1-5-18\15c3da9d-e25c-41ff-97dc-449a0d7d58dd                                                          
C:\Windows\System32\Microsoft\Protect\S-1-5-18\Preferred                                                                                     
C:\Windows\System32\Microsoft\Protect\S-1-5-18\User\3700badb-931e-44b5-b3f5-ea3b232c32d6                                                     
C:\Windows\System32\Microsoft\Protect\S-1-5-18\User\1ede99f5-79b5-418f-9c40-20deb084e219                                                     
C:\Windows\System32\Microsoft\Protect\S-1-5-18\User\eff1de12-4bb7-4760-9dfe-e9626bef6405                                                     
C:\Windows\System32\Microsoft\Protect\S-1-5-18\User\Preferred                                                                                
```

**Comment by Papotito123** [comment]
Hello:
Win 10 1809 x64.
I faced same problem.
I thought it was a bad compiling.
Other commands works well.
It seems more of privileges to access this file.

If there's a more knowledge explanation and how to deal with it, be much appreciated.

Thanks.

**Comment by leftp** [comment]
Files are only accessible by SYSTEM and i think there is a missing Helpers.GetSystem(); in TriageSystemMasterKeys since LSADump.GetDPAPIKeys(true) revs2self

**Comment by Papotito123** [comment]
Hello:
@leftp , how can we deal with that?

Thanks.

**Comment by HarmJ0y** [comment]
Should be fixed with https://github.com/GhostPack/SharpDPAPI/pull/7 . If there's still an issue I can reopen.

**Comment by Papotito123** [comment]
Hello: 
Tested latest git with fix #7 and machinemasterkeys / machinecredentials / machinevaults / machinecerts commands worked very well.

Just one comment.
When running machinetriage , it runs machinecredentials and machinevaults . 
But it also runs machinecerts that produces a very huge list(at least for me ,and I have just  a home computer).

Thanks for the fix.

**Comment by HarmJ0y** [comment]
The certificate enumeration is now intended for machine triage. I'll look into other alias/targeting options.
